### PR TITLE
(test) O3-3853: Expand unit test coverage for the datepicker component

### PR DESCRIPTION
**Pull request by Josephkagimu1**
## Requirements

- [x] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-docs.openmrs.org/en-US/docs/frontend-modules/contributing/#contributing-guidelines) label. See existing PR titles for inspiration.
- [ ] My work is based on designs, which are linked or shown either in the Jira ticket or the description below. (See also: [Styleguide](http://om.rs/o3ui))
- [x] My work includes tests or is validated by existing tests.
- [ ] I have updated the esm-framework and storybook mocks ([mock.tsx](https://github.com/openmrs/openmrs-esm-core/blob/main/packages/framework/esm-framework/mock.tsx), [mock-jest.tsx](https://github.com/openmrs/openmrs-esm-core/blob/main/packages/framework/esm-framework/mock-jest.tsx), and [storybook mocks](https://github.com/openmrs/openmrs-esm-core/blob/main/packages/tooling/storybook/mocks/)) to reflect any API changes.

## Summary
The `OpenmrsDatePicker` component already had baseline test coverage (introduced in #1303 and expanded in #1639). This PR grows the suite from 20 to 34 tests, adding: a first date-selection interaction test verifying `onChange` fires, `maxDate` boundary clamping, `null`/`undefined` value handling, the deprecated `label` prop and `aria-labelledby` support, `short`/`light`/custom `className` CSS variants, and negative cases for the disabled state and callback warnings.

## Related Issue
https://openmrs.atlassian.net/browse/O3-3853

## Other
All Tests pass locally 
No unrelated changes included
Follows existing test patterns in the codebase

### CODE REVIEW

**Comment by Josephkagimu1** [comment]
CC: @pirupius , @jayasanka-sack

**Comment by pirupius** [comment]
@Josephkagimu1 your PR has test failures

**Comment by Josephkagimu1** [comment]
CC : @pirupius , I worked on them

File: packages/framework/esm-styleguide/src/datepicker/datepicker.test.tsx [hunk 1]
```diff
@@ -198,14 +281,45 @@ describe('OpenmrsDatePicker', () => {
...
+      await waitFor(() => {
+        expect(screen.getByRole('dialog')).toBeInTheDocument();
+      });
+      const dialog = screen.getByRole('dialog');
+      const grid = within(dialog).getByRole('grid');
+      const dayCell = within(grid).getByRole('button', { name: /15/ });
+      await user.click(dayCell);
+      await waitFor(() => {
+        expect(onChange).toHaveBeenCalled();
+      });
```
Comment (denniskigen) [inline packages/framework/esm-styleguide/src/datepicker/datepicker.test.tsx:322]: Let's also check *which* date was passed, so a timezone off-by-one can't slip past.

```suggestion
      await waitFor(() => {
        expect(onChange).toHaveBeenCalledWith(new Date(2025, 5, 15));
      });
```

File: packages/framework/esm-styleguide/src/datepicker/datepicker.test.tsx [hunk 2]
```diff
@@ -198,14 +281,45 @@ describe('OpenmrsDatePicker', () => {
...
+    });
```
Comment (denniskigen) [inline packages/framework/esm-styleguide/src/datepicker/datepicker.test.tsx:323]: Nit: the same kind of test for `OpenmrsDateRangePicker` would round this out. If you'd like to include it, this passes as-is when added to the `calendar popover` describe block in `date-range-picker.test.tsx`:

```tsx
it('should select a range and call onChange', async () => {
  const user = userEvent.setup();
  const onChange = vi.fn();
  render(
    <OpenmrsDateRangePicker
      aria-label="datepicker"
      defaultValue={[new Date(2025, 5, 1), new Date(2025, 5, 2)]}
      onChange={onChange}
    />,
  );
  await user.click(screen.getByRole('button'));
  await waitFor(() => {
    expect(screen.getByRole('dialog')).toBeInTheDocument();
  });
  const grid = within(screen.getByRole('dialog')).getByRole('grid');
  await user.click(within(grid).getByRole('button', { name: /10/ }));
  await user.click(within(grid).getByRole('button', { name: /15/ }));
  await waitFor(() => {
    expect(onChange).toHaveBeenCalledWith([new Date(2025, 5, 10), new Date(2025, 5, 15)]);
  });
});
```

Also fine as a follow-up if you'd rather keep this PR focused.

**Review by denniskigen — COMMENTED** [review]
Good work, @Josephkagimu1. This is a solid expansion of the datepicker coverage, and thanks for iterating on the CI failures. One suggestion on the selection test, then this is good to go.

**Comment by Josephkagimu1** [comment]
Hi @denniskigen , I added what you requested , thanks for that review .

**Review by denniskigen — APPROVED** [review]
LGTM! Thanks, @Josephkagimu1. Nice work seeing this through the CI fixes and review iterations.
